File: CS/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public int GetDecimalValue(ListNode head) {
        int result = 0;
        while(head != null){
            result = result << 1;
            result = result | head.val;
            //result += head.val;
            head = head.next;
        }
        return result;
    }
}

File: Cpp/1290_Convert_Binary_Number_in_a_Linked_List_to_Integer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        return head ? getDecimalValue(head->next,result*2+head->val) : result;
        /*
        int result = 0;
        while(head != nullptr){
            result <<= 1;
            //cout << result;
            result |= head->val;
            //cout << result;
            head = head->next;
        }
        return result;
        */
    }
};
